src/index.ts

```diff
@@ -1,5 +1,33 @@
-export default function alwaysArray<T>(input: T | T[]): T[] {
-	return Array.isArray(input) ? input : [input];
+type SingleOrArray<T> = T | T[] | Iterable<T> | readonly T[];
```

Comment:
```suggestion
type SingleOrArray<T> = T | Iterable<T> | readonly T[];
```

`T[]` can be passed to `readonly T[]`